biomage/account/account.py

```diff
@@ -137,19 +140,60 @@ def _create_user(full_name, email, password, userpool, overwrite=False):
     email = email.lower()
 
     error = create_account(full_name, email, userpool)
-    # if the user already exists, just proceed and change the password
-    # this way, when there's an error creating a list you can just
-    # re-run the whole script and get the correct tmp passwords
+    # Return error if we are not overwriting user password
     if error and ("usernameExistsException" not in str(error) or not overwrite):
         return error
 
-    if error:
-        return error
     error = _change_password(email, password, userpool)
     if error:
         return error
 
 
+def _validate_input(email, full_name):
+    """
+    Check if the information provided for user creation is valid:
+    - Email is valid (according to regex in https://emailregex.com/)
+    - Full name is provided
+    Returns error message if any of the checks fail
+    """
+    if not email or pd.isna(email):
+        return f"ERROR: Email not provided for user {full_name}"
+
+    if not full_name or pd.isna(full_name):
+        return f"ERROR: Full name not provided for user {email}"
+
+    if not re.match(r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", email):
+        return f"ERROR: Email {email} does not match regex"
+
+
+def _user_exists(email, userpool):
```

Comment:
this function is not needed at all, all logic is already handled in line 143

Reply:
The reasoning I take out the logic on line 143 is so that the `.out` file can better resemble the fact that we do not create new accounts for users that already exist. Alternatively, I can use the logic on line 143 to write the `.out` file after the user creation. This might not be a good idea because there might be a case where the user account is created, but an error happens before writing the password into an`.out` file. What do you think?

Reply:
You can just do the following and get rid of the extra function and code in 256. In general, try to keep this code simple. Running this script twice is much better than making it too complex. If we create a new account and we don't save the password it does not matter, we can just change it with a single command again, or overwrite them all. However, those cases are already covered in the code below anyway

```
            error = _create_user(full_name, email, password, userpool, overwrite)
            if "usernameExistsException" in str(error):
                print(f"{email} already exists in the user pool skipping creation.")
                out.write("%s,%s,'Already have an account'\n" % (full_name, email))
            if error:
                raise Exception(error)

            print("%s,%s,%s" % (full_name, email, password))
            out.write("%s,%s,%s\n" % (full_name, email, password))
```